src/biteme/__init__.py

```diff
@@ -1,5 +1,88 @@
-from biteme.__main__ import cli
-from biteme.pybites import _bite_info, download_bite
+from __future__ import annotations
 
+import cgi
+import io
+import operator
+import os
+import pathlib
+import zipfile
+from typing import Any
+from typing import Union
 
-__all__ = ["cli", "download_bite", "_bite_info"]
+import pydantic
+import requests
+import typer
+
+
+class BiteInfo(pydantic.BaseModel):
+    class Config:
+        allow_mutation = False
+
+    number: int
+    title: str
+    description: str
+    level: str
+    tags: frozenset[str]
+    free: bool
+    score: int
+    function: str
+
+
+def get_bite_info(number: int) -> BiteInfo:
+    """Get info for a specific bite.
+
+    Example:
+        >>> bite = get_bite_info(1)
+
+    """
+    url = f"https://codechalleng.es/api/bites/{number}"
+    response = requests.get(url)
+    response.raise_for_status()
+    [data] = response.json()
+    return BiteInfo(**data)
+
+
+def get_all_bite_info() -> list[BiteInfo]:
+    """Get info for all the bites.
+
+    Example:
+        >>> bites = get_all_bite_info()
+
+    """
+    url = "https://codechalleng.es/api/bites/"
```

Comment:
`url` is defined multiple times, could put that in a constant

Reply:
I could for the prefix, yeah. I suppose the URLs aren't clickable because they have format strings, so no harm in pulling it out.

Reply:
Yeah I try to deduplicate as much as possible